### PR TITLE
Medical - Add Alternative Survival Chance

### PR DESCRIPTION
**Pull request by BrettMayson**
Co-authored-by: @TyroneMF

Alternative to #8170

**When merged this pull request will:**
- Adds a percentage slider for surviving a fatal injury.

### CODE REVIEW

**Comment by BrettMayson** [comment]
I am aware as soon as I pushed that AI is now invincible. I am declaring this a feature, not a bug.

File: addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf [hunk 1]
```diff
@@ -122,8 +122,10 @@ private _bodyPartVisParams = [_unit, false, false, false, false]; // params arra
                 _critialDamage = true;
             };
             if ([_unit, _bodyPartNToAdd, _bodyPartDamage, _woundDamage] call FUNC(determineIfFatal)) then {
-                TRACE_1("determineIfFatal returned true",_woundDamage);
-                [QEGVAR(medical,FatalInjury), _unit] call CBA_fnc_localEvent;
+                if !(isPlayer _unit && {random 1 > EGVAR(medical,deathChance)}) then {
```
Comment (jonpas) [inline addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]: `true && true -> !true -> false`, so if player and death chance above, it won't cause fatal injury? This seems reversed.

Comment (kymckay) [reply addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]: De Morgan's law, the code as is can be simplified to:

!isPlayer || random 1 <= deathChance

**Comment by kymckay** [comment]
Even though it's effectively the same as #8170 but inverse logic, it does feel like this makes more sense intuitively and if there's demand for it I do prefer this method to the other.

File: addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf (hunk 1 above)
Comment (jonpas) [reply addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]: You are correct, I was still reading it as survivability chance. If death chance is 0.15 then random value has to be below that, not above, so you get `true && false` when it should happen actually. :D

I would prefer inverting that with De Morgan's law as you state, easier to read. But that may just be me still stuck with "survival chance" making more sense in code, but probably not to the user.

**Comment by Italiandrache** [comment]
> I am aware as soon as I pushed that AI is now invincible. I am declaring this a feature, not a bug.

What do you mean?

**Comment by BrettMayson** [comment]
> 
> 
> > I am aware as soon as I pushed that AI is now invincible. I am declaring this a feature, not a bug.
> 
> What do you mean?

It was a joke since the first commit had that oversight, it's good now

File: addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf (hunk 1 above)
Comment (jonpas) [reply addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]: @BrettMayson Can we reverse that? Then I am happy to merge.

Comment (BrettMayson) [reply addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]: Yeah sure I can invert it

**Review by jonpas — APPROVED** [review]

File: addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf [hunk 2]
```diff
@@ -122,8 +122,10 @@ private _bodyPartVisParams = [_unit, false, false, false, false]; // params arra
                 _critialDamage = true;
             };
             if ([_unit, _bodyPartNToAdd, _bodyPartDamage, _woundDamage] call FUNC(determineIfFatal)) then {
-                TRACE_1("determineIfFatal returned true",_woundDamage);
-                [QEGVAR(medical,FatalInjury), _unit] call CBA_fnc_localEvent;
+                if (!isPlayer || random 1 <= EGVAR(medical,deathChance)) then
```
Comment (veteran29) [inline addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]:
```suggestion
                if (!isPlayer _unit || random 1 <= EGVAR(medical,deathChance)) then
```

Comment (BrettMayson) [reply addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]: ah lol, thanks. I just copy pasted from the comment on my phone and didn't notice

File: addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf [hunk 3]
```diff
@@ -122,7 +122,7 @@ private _bodyPartVisParams = [_unit, false, false, false, false]; // params arra
                 _critialDamage = true;
             };
             if ([_unit, _bodyPartNToAdd, _bodyPartDamage, _woundDamage] call FUNC(determineIfFatal)) then {
-                if !(isPlayer _unit && {random 1 > EGVAR(medical,deathChance)}) then {
+                if (!isPlayer || random 1 <= EGVAR(medical,deathChance)) then
```
Comment (kymckay) [inline addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf]:
```suggestion
                if (!isPlayer || random 1 < EGVAR(medical,deathChance)) then
```

Very minor thing, but something I alluded to in my other comment. Arma's random operation is exclusive of the upper bound and inclusive of the lower which is 0, so for the setting to be the true chance the comparison should just be less than (consider the edge cases and it'll become clear why).

Comment (jonpas) [reply addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf]: Added missing opening brace too.

File: addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf [hunk 4]
```diff
@@ -122,8 +122,10 @@ private _bodyPartVisParams = [_unit, false, false, false, false]; // params arra
                 _critialDamage = true;
             };
             if ([_unit, _bodyPartNToAdd, _bodyPartDamage, _woundDamage] call FUNC(determineIfFatal)) then {
-                TRACE_1("determineIfFatal returned true",_woundDamage);
-                [QEGVAR(medical,FatalInjury), _unit] call CBA_fnc_localEvent;
+                if (!isPlayer _unit || {random 1 <= EGVAR(medical,deathChance)}) then {
```
Comment (kymckay) [inline addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf:125]:
```suggestion
                if (!isPlayer _unit || {random 1 < EGVAR(medical,deathChance)}) then {
```

**Review by veteran29 — APPROVED** [review]

**Review by kymckay — APPROVED** [review]

**Pull request merged**